openlibrary/core/ia.py

```diff
@@ -292,3 +298,99 @@ def add_subjects(self):
             self['subjects'] = subjects
+
+_ia_db = None
+def get_ia_db(configfile=None):
+    """Metadata API is slow.
+
+    Talk to archive.org database directly if it is specified in the
+    global configuration or if a configfile is provided.
+    """
+    if configfile:
+        from openlibrary.config import load_config
+        load_config(configfile)
+
+    if not config.get("ia_db"):
+        return None
+    global _ia_db
+    if not _ia_db:
+        settings = config.ia_db
+        host = settings['host']
+        db = settings['db']
+        user = settings['user']
+        pw = os.popen(settings['pw_file']).read().strip()
+        _ia_db = web.database(dbn="postgres", host=host, db=db, user=user, pw=pw)
+    return _ia_db
+
+
+def get_candidate_ocaids(since_days=None, since_date=None,
+                         scanned_within_days=60, repub_states=None,
+                         marcs=True, lazy=False, count=False, db=None):
+    """Returns a list of identifiers which match the specified criteria
+    and are candidates for ImportBot.
+
+    If since_days and since_date are None, no date restrictions are
+    applied besides `scanned_within_days` (which may also be None)
+
+    Args:
+        since_days (int) - number of days to look back for item updates
+        since_date (Datetime) - only completed after since_date
+                                (default: if None, since_days used)
+        scanned_within_days (int) - only consider items `scanned_within_days`
+                                    of `since_date` (default 60 days; 2 months)
+        marcs (bool) - require MARCs present?
+        lazy (bool) - if True, returns query as iterator, otherwise,
+                      returns identifiers list
+        count (bool) - if count, return (long) number of matching identifiers
+        db (web.db) - A web.py database object with an active
+                      connection (optional)
+
+    Usage:
+        >>> from openlibrary.core.ia import get_ia_db, get_candidate_ocaids
+        >>> candidates = get_candidate_ocaids(
+        ...    since_days=1, db=get_ia_db('openlibrary/config/openlibrary.yml'))
+
+    Returns:
+        A list of identifiers which are candidates for ImportBot, or
+        (if count=True) the number of candidates which would be returned.
+
+    """
+    db = db or get_ia_db()
+    date = since_date or date_n_days_ago(n=since_days)
+    min_scandate = date_n_days_ago(start=date, n=scanned_within_days)
+    repub_states = repub_states or VALID_READY_REPUB_STATES
+
+    qvars = {
+        'c1': '%opensource%',
+        'c2': '%additional_collections%'
+    }
+
+    _valid_repub_states_sql = "(%s)" % (', '.join(str(i) for i in repub_states))
+    q = (
+        "SELECT " + ("count(identifier)" if count else "identifier") + " FROM metadata" +
+        " WHERE repub_state IN " + _valid_repub_states_sql +
+        "   AND mediatype='texts'" +
+        "   AND (noindex IS NULL OR collection='printdisabled')" +
```

Comment:
Hm, I wondered this too. I talked to Alexis and she suggested that printdisabled is a superset of anything that would be inlibrary.

I remember asking about `lendinglibrary`, I think the answer was that they would not be no-index or would be in printdisabled as well. I can check w/ Alexis again.